### DISCUSSION TITLE
How to provide a geojson from postgrest?

### DISCUSSION THREAD

**Discussion by npltr62** [PostgREST/postgrest#1894] [Q&A] [answered]
### Environment

* PostgreSQL version: kartoza/postgis:13
* PostgREST version: postgrest:v7.0.1
* Operating system:
NAs synology

### Description of issue
I would like to get a geojson with featurecollection
I try to set accept parameter like that 'Accept:  "application/geo+json" but nothing works.

**Comment by steve-chavez** [comment, marked as answer]
Hey @npltr62,

> I try to set accept parameter like that 'Accept: "application/geo+json" but nothing works.

Yeah, this feature is not yet merged: https://github.com/PostgREST/postgrest/pull/1564

For now you can create a view for getting geojson, like:

```sql
create view shops_geojson as
select st_asgeojson(x) from shops x;
```

Then call it through PostgREST:

```http
GET /shops_geojson


{"type": "Feature", 
 "geometry": {"type":"Point","coordinates":[-71.10044,42.373695]}, 
 "properties": {"id": 1, "address": "1369 Cambridge St"}
} 
```

**Edit**: For a feature collection

```sql
create view shops_geojson as
select json_build_object('type', 'FeatureCollection', 'features', json_agg(ST_AsGeoJSON(x)))  from shops x;
```

**Reply by npltr62** [reply]
ok thanks. and for getting feature collection?

**Reply by wolfgangwalther** [reply]
@steve-chavez ping. Any idea? I have no clue about postgis.

**Reply by steve-chavez** [reply]
Sorry, missed this one. I've edited my answer above, took the function from https://github.com/PostgREST/postgrest/pull/1564/files#diff-6600cf9df2241e8fc293e9d89997c585750e0405f235e3dfb282d0c2e8a44a02R88
